# 61) Bağımlı Örneklem T Testi (Eşleştirilmiş/Paired T Test)
Bağımsız İki Örneklem T Testi'nde (Konu 58), iki grup **birbirinden tamamen farklı, ilişkisiz bireylerden** oluşuyordu. Bağımlı Örneklem T Testi ise, **aynı bireylerin** iki farklı zamanda/koşulda ölçüldüğü durumları test eder.

## Ne Zaman Kullanılır?
İki ölçüm, **birbiriyle eşleştirilebilir/ilişkilidir.** En yaygın senaryolar:
- **Öncesi-Sonrası (Before-After):** Aynı kişilerin bir müdahale/kampanya öncesi ve sonrası ölçümü (örn: "yeni pazarlama stratejisi öncesi ve sonrası aynı müşterilerin harcaması")
- **Eşleştirilmiş Çiftler:** İki farklı ama birbiriyle eşleştirilmiş gözlem (örn: ikiz kardeşler, aynı ürünün iki farklı ölçümü)

## Bağımsız'dan Temel Farkı
Bağımsız testte, iki grup **ayrı bireylerden** oluştuğu için varyanslar ayrı ayrı ele alınıyordu (Pooled/Welch). Bağımlı testte ise, aynı kişiler tekrar ölçüldüğü için, **iki ölçüm arasındaki FARKLARA** odaklanırız bu, problemi aslında **Tek Örneklem T Testi'ne (Konu 58) indirger!**

## Mantık (Neden Tek Örneklem Testine Benzer)
Her bireyin "önce" ve "sonra" değerleri arasındaki **farkı** ($d_i = x_{sonra,i} - x_{once,i}$) hesaplarız. Sonra bu farkların ortalamasının, **0'dan farklı olup olmadığını** test ederiz — tıpkı Konu 58'de "ortalama, belirli bir değere eşit mi" sorusunu sorduğumuz gibi, burada "fark ortalaması 0'a eşit mi" sorusunu soruyoruz.

## Hipotezler
- **H0:** $\mu_d = 0$ (öncesi-sonrası arasında ortalama fark yoktur)
- **H1:** $\mu_d \ne 0$ (fark vardır) — ya da tek yönlü versiyonu

## Test İstatistiği Formülü
$$t = \frac{\bar{d} - 0}{s_d / \sqrt{n}}$$

Burada $\bar{d}$, fark değerlerinin ortalaması, $s_d$ ise fark değerlerinin standart sapması. **Formül, Tek Örneklem T Testi'yle 
(Konu 57) birebir aynı** sadece ham veriler yerine, "farklar" kullanılıyor.

## Neden Bağımsız Test Yerine Bunu Kullanmalıyız?
Aynı kişileri iki kere ölçtüğümüzde, bu ölçümler **birbirinden bağımsız değildir** (bir kişinin "önce" değeri yüksekse, "sonra" değeri de muhtemelen yüksek çıkar — kişisel farklılıklar iki ölçümde de korunur). Bağımsız test kullanmak, bu doğal ilişkiyi yok sayar ve **gücü (Power, Konu 52) azaltır.** Bağımlı test, bu kişisel varyasyonu "farkları alarak" ortadan kaldırır, sadece **gerçek değişime** odaklanır — bu yüzden aynı kişiler tekrar ölçüldüğünde her zaman bağımlı test tercih edilmelidir.

## Python'da Kullanımı
```python
from scipy import stats
t_stat, p_value = stats.ttest_rel(once, sonra)
```
`ttest_rel` — "related" (ilişkili) kelimesinden geliyor.

In [6]:
import numpy as np
from scipy import stats
import pingouin as pg

# Senaryo: 10 müşterinin, yeni sadakat programı ÖNCESİ ve SONRASI
# aylık ortalama harcaması (TL). AYNI müşteriler iki kere ölçülüyor
oncesi = np.array([420, 380, 510, 350, 460, 400, 390, 440, 470, 410])
sonrasi = np.array([465, 410, 540, 375, 495, 425, 430, 480, 500, 445])

# biz yeni sadakat programının ortalama harcamayı artıracağını düşünüyoruz.
# Hipotezleri Kuralım.
# H0: Yeni sadakat programı öncesi harcamaların ortalaması ile yeni sadakat programı öncesi harcamaların ortalaması aynıdır.
# H1: Yeni sadakat programı sonrası harcamaların ortalaması, yeni sadakat programı öncesi harcamaların ortalamasından fazladır.
farklar = np.mean(sonrasi) - np.mean(oncesi)
print(f'Sadakat programı sonrası - sadakat programı öncesi = {farklar}')
ttest = stats.ttest_rel(sonrasi,oncesi, alternative='greater')
alpha = 0.05
p_degeri = round(ttest.pvalue,5)
df = ttest.df
t_istatistigi = ttest.statistic
low, high = ttest.confidence_interval(confidence_level=0.95)
print(f'P Değeri: {p_degeri}')
print(f'Serbestlik Derecesi: {df}')
print(f'T İstatistiği: {t_istatistigi}')
print(f'%95 güven düzeyi için alt sınır: {low}, üst sınır: {high}')
if p_degeri < alpha:
    print('H0 reddedilir, yeni sadakat programı sonrası harcamaların ortalaması, yeni sadakat programı öncesi harcamaların ortalamasından fazladır.')
else:
    print('H0 reddedecek kadar elimizde güçlü bir kanıt yok, Yeni sadakat programı öncesi harcamaların ortalaması ile yeni sadakat programı sonrası harcamaların ortalaması aynıdır.')

d_degeri = pg.compute_effsize(sonrasi, oncesi, eftype='cohen')
print("Cohen's d:", d_degeri)

Sadakat programı sonrası - sadakat programı öncesi = 33.5
P Değeri: 0.0
Serbestlik Derecesi: 9
T İstatistiği: 15.841019165405925
%95 güven düzeyi için alt sınır: 29.62340073559842, üst sınır: inf
H0 reddedilir, yeni sadakat programı sonrası harcamaların ortalaması, yeni sadakat programı öncesi harcamaların ortalamasından fazladır.
Cohen's d: 0.6934127084005438


### Sonuç
10 Yeni müşterinin sadakat programı öncesi ve sonrası harcamaları arasındaki farkın istatistiksel olarak anlamlı olup olmadığını belirlemek üzere yaptığımız bağımlı örneklem *t* testi sonuçlarına göre, sadakat programı sonrası harcamaların ortalaması, sadakat programı öncesi harcamaların ortalamasından 33.5 tl anlamlı olarak fazladır (p<0.05). Bu durumda yeni sadakat programının harcamaları artırdığı söylenebilir. Ayrıca yeni sadakat programının harcamalar üzerinde orta-büyük düzeyde bir etkisi olduğu söylenebilir (|d|=0.69).